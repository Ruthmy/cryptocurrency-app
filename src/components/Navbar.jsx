import React from 'react';
import { NavLink } from 'react-router-dom';

import '../styles/Navbar.css';

const Navbar = () => (
  <>
    <nav>
      <NavLink to="/">
        <div className="nav__logo d-flex-row">
          <img src="logo192.png" alt="Space Travelers Logo" />
          <h1>Space Travelers&apos; Hub</h1>
        </div>
      </NavLink>
      <div>
        <ul className="nav__routes d-flex-row">
          <li key="Rockets">
            <NavLink
              to="/"
              style={({ isActive }) => ({
                textDecoration: isActive ? 'underline' : 'none',
              })}
            >
              Rockets
            </NavLink>
          </li>
          <li key="Missions">
            <NavLink
              to="/missions"
              style={({ isActive }) => ({
                textDecoration: isActive ? 'underline' : 'none',
              })}
            >
              Missions
            </NavLink>
          </li>
          <p>|</p>
          <li key="My Profile">
            <NavLink
              to="/profile"
              style={({ isActive }) => ({
                textDecoration: isActive ? 'underline' : 'none',
              })}
            >
              My Profile
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
    <div className="nav__line" />
  </>
);

export default Navbar;
