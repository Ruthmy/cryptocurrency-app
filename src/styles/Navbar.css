nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 4rem;
  width: 100%;
  height: 8rem;
  padding: 6rem 6rem;
  background: rgba(255, 255, 255, 0.3);
  backdrop-filter: blur(15px);
  -webkit-backdrop-filter: blur(15px);

  /* box-shadow: 0 4rem 2rem 4rem rgba(0, 0, 0, 0.03); */
}

.nav__logo {
  align-items: center;
  color: rgb(0, 0, 0);
}

.nav__logo img {
  width: 13%;
  margin-right: 2rem;
}

.nav__logo h1 {
  font-weight: 500;
}

.nav__line {
  border-bottom: 2px solid rgb(224, 224, 224);
  height: 2px;
  margin: 0 4rem;
}

.nav__routes {
  gap: 3rem;
}

.nav__routes a {
  color: rgb(1, 123, 255);
}
