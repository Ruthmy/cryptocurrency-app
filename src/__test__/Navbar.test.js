import React from 'react';
import { render, screen } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';

import Navbar from '../components/Navbar';

// Helper function to wrap the component in HashRouter for testing NavLink
const renderWithRouter = (component) => render(<HashRouter>{component}</HashRouter>);

describe('Navbar component', () => {
  test('Should have the home link with it icon', () => {
    renderWithRouter(<Navbar />);
    const homeLinkElement = screen.getByText('home');
    const homeIconElement = screen.getByAltText('Back');
    expect(homeLinkElement).toBeInTheDocument();
    expect(homeIconElement).toBeInTheDocument();
  });

  test('Should have the buttons images', () => {
    renderWithRouter(<Navbar />);
    const MicrophoneButton = screen.getByAltText('Microphone');
    const SettingsButton = screen.getByAltText('Settings');
    expect(MicrophoneButton).toBeInTheDocument();
    expect(SettingsButton).toBeInTheDocument();
  });

  test('Should render correctly', () => {
    renderWithRouter(<Navbar />);
    const { container } = renderWithRouter(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
