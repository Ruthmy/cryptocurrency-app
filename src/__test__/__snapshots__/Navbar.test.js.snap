// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Should render correctly 1`] = `
<div>
  <nav
    class="d-flex-row"
  >
    <div>
      <a
        aria-current="page"
        class="active"
        href="#/"
      >
        <img
          alt="Back"
          class="nav__back"
          src="arrow.svg"
        />
        home
      </a>
    </div>
    <div
      class="nav__panel d-flex-row"
    >
      <button
        class="btn"
        type="button"
      >
        <img
          alt="Microphone"
          src="microphone.svg"
        />
      </button>
      <button
        class="btn"
        type="button"
      >
        <img
          alt="Settings"
          src="gear.svg"
        />
      </button>
    </div>
  </nav>
</div>
`;
